### PR TITLE
Apply QC filters to a MT, output a VCF for associaTR 

### PR DESCRIPTION
**Pull request by hopedisastro**
This script applies filters to a ExpansionHunter MT, and outputs a VCF ready for input into associaTR.
Input MT should be the output of qc_annotator.py

Applied filters:
- remove monomorphic variants
- locus-level callrate >=0.9
- sample-level call rate >=0.99
- observed heterozygosity >=0.00995 (corresponds to a MAF = 0.5%)
- locus-level HWEP (binom definition) >=10^-6
- set calls outside of [-30,20] relative to mode to NA
- enforce locus-level call rate >=0.9

### CODE REVIEW

**Comment by MattWellie** [comment]
@hopedisastro happy to have a look at this, but as a shortcut, if you give hail a `...vcf.bgz` path to export to, it writes a block-gzip'd file, and the argument `tabix=true` to the export_vcf method will also generate an index, so you can happily lose a lot of code and arguments here

**Comment by hopedisastro** [comment]
Thanks Matt, will definitely re-organise it a bit

File: str/qc/qc_filters_associatr.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        (mt.num_alleles > 1)
+        & (mt.variant_qc.call_rate >= 0.9)
+        & (mt.obs_het >= 0.00995)
+        & (mt.binom_hwep >= 0.000001)
+    )
+
+    # set sample level call rate >=0.99
+    mt = mt.filter_cols(mt.sample_qc.call_rate >= 0.99)
+
+    # set big expansions/deletions beyond [-30,20] relative to mode allele to NA
```
Comment (MattWellie) [inline str/qc/qc_filters_associatr.py:58]: you could probably squash all this syntax down... something like

```
condition = (mt.allele_1_minus_mode > 20) | (mt.allele_1_minus_mode < -30)
mt = mt.annotate_entries(
        GT=hl.if_else(condition, hl.missing('call'), mt.GT),
        ADFL=hl.if_else(condition, hl.missing('str'), mt.ADFL),
        ....
```

File: str/qc/qc_filters_associatr.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        'allele_2_bp_length',
+        'allele_1_REPCI_over_CN',
+        'allele_2_REPCI_over_CN',
+        'allele_1_minus_mode',
+        'allele_2_minus_mode',
+        'allele_1_REPCI_size',
+        'allele_2_REPCI_size',
+        'allele_1_is_non_mode',
+        'allele_2_is_non_mode',
+    )
```
Comment (MattWellie) [inline str/qc/qc_filters_associatr.py:169]: writing to a VCF will drop everything except locus, alleles, quality, id, filter, format, and GT. Hail will drop everything else automatically

File: str/qc/qc_filters_associatr.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    mt = mt.annotate_entries(QUAL=hl.missing('float64'))
+
+    # Set 'rsid' to REPID
+    mt = mt.annotate_rows(rsid=mt.REPID)
+
+    # Key by locus and alleles
+    mt = mt.key_rows_by('locus', 'alleles')
+
+    # Remove 'CPG' prefix from ID (associaTR expects ID to be numeric)
+    mt = mt.annotate_cols(s_no_prefix=mt.s[3:])
```
Comment (MattWellie) [inline str/qc/qc_filters_associatr.py:181]:
```suggestion
    mt = mt.annotate_cols(s_no_prefix=mt.s.replace("^CPG([0-9])", "$1"))
```

ignore this, I just poked around the hail docs and thought this syntax was cool

**Review by MattWellie — APPROVED** [review]
verbose, but should be good!

**Pull request merged**